### PR TITLE
add function to retrieve className from task and test. fixes #829

### LINKED ISSUE #829: Anonymous task implementation results in (null) on start-up log
When I do

    environment.admin().addTask(new Task("foo"){//implement execute//})

instead of implementing a concrete Task sub-class, I end up with

    INFO  [2015-01-01 15:22:44,045] io.dropwizard.setup.AdminEnvironment: tasks = 
        POST    /tasks/foo (null)

on server-start-up.

"null" normally would be the implementing class name. 

I guess, since it it just an info, a possible solution would be to just leave out the (null) part in that case.

Update: I implemented a solution that falls back to getClass().getName() when getCanonicalName() is null, so the log displays

        POST /tasks/foo (my.implementing.Class$1)

which allows me to identify the code where the Task is implemented.

### PR DESCRIPTION
**Pull request by jangalinski**
instead of task.getClass().getCanonicalName(), task.getClass().getName is used for anonymous implementations of task.

### CODE REVIEW

File: dropwizard-core/src/test/java/io/dropwizard/setup/AdminEnvironmentTest.java [hunk 1]
```diff
@@ -11,6 +11,7 @@
 import javax.servlet.ServletRegistration;
 import java.io.PrintWriter;
 
+import static org.assertj.core.api.Assertions.*;
```
Comment (joschi) [inline dropwizard-core/src/test/java/io/dropwizard/setup/AdminEnvironmentTest.java]: Please don't use wildcard imports.

File: dropwizard-core/src/main/java/io/dropwizard/setup/AdminEnvironment.java [hunk 2]
```diff
@@ -59,11 +72,13 @@ public void addTask(Task task) {
...
         final StringBuilder stringBuilder = new StringBuilder(1024).append(String.format("%n%n"));
 
+
         for (Task task : tasks.getTasks()) {
+
             stringBuilder.append(String.format("    %-7s /tasks/%s (%s)%n",
                                                "POST",
                                                task.getName(),
-                                               task.getClass().getCanonicalName()));
+                                               getTaskClassName.apply(task)));
```
Comment (joschi) [inline dropwizard-core/src/main/java/io/dropwizard/setup/AdminEnvironment.java]: I'd simply use [´MoreObjects#firstNonNull(T, T)`](http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/base/MoreObjects.html#firstNonNull(T,%20T)) in this case. No need for this `Function<Task, String>`.

**Comment by jangalinski** [comment]
I wanted to keep the change testable, so i went for an extra function.  But firstnonnull is a good idea, no need to test guava here.   Will refactor and commut again.

**Comment by jangalinski** [comment]
Question: why not just use getClass().getName()? No null-checks, no functions, just the class name, and done.
I mean, its only logging and when inner classes are represented with i.d.Bar$Foo instead i.d.Bar.Foo, who cares?

**Comment by joschi** [comment]
Thanks for your PR!

**Pull request closed without merging**